### PR TITLE
M7 dcache ctrl via a parameter

### PR DESCRIPTION
**Pull request by davids5**
The FORCE_F7_DCACHE parameter can be set to
   0 - (default) if the Errata exist turn dcache off else leave it on
   1 -  Force it off
   2 -  Force it on

   At boot the system will disable the d-cache if the silicon
   has the 1259864 Data corruption in a sequence of Write-Through
   stores and loads eratta.

   Post nsh script execution the FORCE_F7_DCACHE parameter
   will be used to set the d-cache to the state indicated
   above.

Added status and control command
```
nsh>dcache
 INFO  [dcache] M7 cpuid r1p0 has dcache eratta! dcache Off

dcache on
 INFO  [dcache] M7 cpuid r1p0 has dcache eratta! dcache On

dcache off
 INFO  [dcache] M7 cpuid r1p0 has dcache eratta! dcache Off

```

### CODE REVIEW

File: src/drivers/stm32/armv7-m_dcache/params.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * a sequence of Write-Through stores and loads on ARM M7 silicon
+ * Fault Status: Present in r0p1, r0p2, r1p0 and r1p1. Fixed in r1p2
+ *
+ *
+ * @min 0
+ * @max 2
+ * @value 0 if Eratta exits turn dcache off else leave it on
+ * @value 1 Force it off
+ * @value 2 Force it on
+ * @group Chip
```
Comment (dagar) [inline src/drivers/stm32/armv7-m_dcache/params.c]:
```suggestion
 * @group Chip
 * @category Developer

```

File: src/drivers/stm32/armv7-m_dcache/params.c [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ */
+PARAM_DEFINE_INT32(FORCE_F7_DCACHE, 0);
```
Comment (dagar) [inline src/drivers/stm32/armv7-m_dcache/params.c]: Need to think of where to put this prefix-wise. Maybe `SYS_`?

**Comment by dagar** [comment]
The build is failing because of the circular dependency between board common and the armv7-m_dcache module.

**Comment by davids5** [comment]
@dagar - OK to come in now?

**Comment by bys1123** [comment]
@davids5 Is this a very rare errata? Seems there is no crash report about this? Is this randomly happen? Or if I get a chip with errata, it will crash at beginning?
Sorry to take up your time, not very clear about this.

**Comment by davids5** [comment]
@bys1123 Please see https://github.com/PX4/Firmware/pull/11374 it is a verified ARM M7 issue. It is present then the STM32F7 - it is non deterministic problem. The precaution to disable d-cache has been taken to error on the side of safety. This PR is to allow using d-cache if one is willing to take a risk of unknown probability.

**Pull request merged**